src/test/java/edu/harvard/iq/dataverse/DatasetTest.java

```diff
@@ -49,5 +56,187 @@ public void testLocksManagement() {
         assertFalse( sut.isLocked() );
         
     }
+
+    // ****************************************************************************************
+    // The following tests test the isDeaccessioned method aiming for 100% prime path coverage.
+    // The following tool [1] was used to calculate the needed test paths.
+    // 11 test paths (= 11 test methods) are needed.
+    // 4 different dataset versions are needed:
+    // - DEACCESSIONED
+    // - DRAFT
+    // - RELEASED
+    // - non of the above (= ARCHIVED)
+    //
+    // See details here: https://github.com/IQSS/dataverse/pull/5703
+    //
+    // [1] https://cs.gmu.edu:8443/offutt/coverage/GraphCoverage
+    // ****************************************************************************************
+
+    private DatasetVersion archivedVersion;
+    private DatasetVersion deaccessionedVersion;
+    private DatasetVersion draftVersion;
+    private DatasetVersion releasedVersion;
+
+    @Before
+    public void before() {
+        this.archivedVersion = new DatasetVersion();
+        this.archivedVersion.setVersionState(VersionState.ARCHIVED);
+
+        this.deaccessionedVersion = new DatasetVersion();
+        this.deaccessionedVersion.setVersionState(VersionState.DEACCESSIONED);
+
+        this.draftVersion = new DatasetVersion();
+        this.draftVersion.setVersionState(VersionState.DRAFT);
+
+        this.releasedVersion = new DatasetVersion();
+        this.releasedVersion.setVersionState(VersionState.RELEASED);
+    }
+
+    @After
+    public void after() {
+        this.archivedVersion = null;
+        this.deaccessionedVersion = null;
+        this.draftVersion = null;
+        this.releasedVersion = null;
+    }
+
+    @Test
+    public void testIsDeaccessionedWithNoVersions() {
```

Comment:
The title of these tests could be a bit clearer. Maybe "testNotDeaccessionedWithNoVersions". I'm up for suggestion, but these tests imply to me that you are looking for cases where isDeaccessioned is true.

Reply:
I agree. The wording is misleading. I changed it with commit 82405bb.
Examples are:
- `datasetShouldNotBeDeaccessionedWithNoVersions`
- `datasetShouldBeDeaccessionedWithDeaccessionedAndArchivedVersions`